### ISSUE TITLE
Navigating with the keyboard between radio-button-groups whose radio-button's set their own name attribute directly is not prevented

### ISSUE THREAD

**Issue by eriklharper** [Esri/calcite-design-system#10849]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

When rendering more than 1 `calcite-radio-button-group` on a page where each individual `calcite-radio-button` is setting their own `name` attribute to the same value within a group (as opposed to setting `name` on `calcite-radio-button-group` instead), using the arrow keys on the keyboard will move between separate groups, treating them like they're all part of the same group when they're not.

### Expected Behavior

Regardless if `name` is set at the parent level or on each individual radio-button, navigating the options with the keyboard should be constrained to just the options in the group.

### Reproduction Sample

https://codepen.io/eriklharper/pen/OJKKQbX?editors=1000

### Reproduction Steps

See the codepen for instructions.

### Reproduction Version

2

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

Calcite (dev)

**Comment by jcfranco** [comment]
This use case is invalid. `name` is required on `calcite-radio-button-group` and when using `calcite-radio-button-group`, the assumption is that all children `calcite-radio-button`s are controlled by the parent (see https://codepen.io/jcfranco/pen/pvzdgzE?editors=1000). 

It may be helpful to include `name` on both [overview](https://developers.arcgis.com/calcite-design-system/components/radio-button/#overview) [sections](https://developers.arcgis.com/calcite-design-system/components/radio-button-group/#overview) and [keyboard](https://developers.arcgis.com/calcite-design-system/components/radio-button/#keyboard-navigation) [sections](https://developers.arcgis.com/calcite-design-system/components/radio-button-group/#keyboard-navigation) to make this clear. cc @geospatialem

**Comment by geospatialem** [comment]
> This use case is invalid. [`name` is required](https://developers.arcgis.com/calcite-design-system/components/radio-button-group/#api-reference-properties-name) on `calcite-radio-button-group` and when using a group, the assumption is that all children `calcite-radio-button`s are controlled by the parent (see https://codepen.io/jcfranco/pen/pvzdgzE?editors=1000).
> 
> It may be helpful to include `name` on both [overview](https://developers.arcgis.com/calcite-design-system/components/radio-button/#overview) [sections](https://developers.arcgis.com/calcite-design-system/components/radio-button-group/#overview) and [keyboard](https://developers.arcgis.com/calcite-design-system/components/radio-button/#keyboard-navigation) [sections](https://developers.arcgis.com/calcite-design-system/components/radio-button-group/#keyboard-navigation) to make this clear. cc @geospatialem @DitwanP

We have an overarching section on [Core concepts](https://developers.arcgis.com/calcite-design-system/core-concepts/#forms) @DitwanP, do you think we should add context to the `name` prop in the "Overview" section? It looks like it's already covered in the keyboard section, so maybe we just need to add the context to the "Overview"?

**Comment by eriklharper** [comment]
> This use case is invalid. [`name` is required](https://developers.arcgis.com/calcite-design-system/components/radio-button-group/#api-reference-properties-name) on `calcite-radio-button-group` and when using a group, the assumption is that all children `calcite-radio-button`s are controlled by the parent (see https://codepen.io/jcfranco/pen/pvzdgzE?editors=1000).
> 
> It may be helpful to include `name` on both [overview](https://developers.arcgis.com/calcite-design-system/components/radio-button/#overview) [sections](https://developers.arcgis.com/calcite-design-system/components/radio-button-group/#overview) and [keyboard](https://developers.arcgis.com/calcite-design-system/components/radio-button/#keyboard-navigation) [sections](https://developers.arcgis.com/calcite-design-system/components/radio-button-group/#keyboard-navigation) to make this clear. cc @geospatialem @DitwanP

Makes sense now that the radio components work differently than the native browser radio inputs.  We can close this!
